cg/store/api/import_func.py

```diff
@@ -106,69 +129,72 @@ def prices_are_same(first_price, second_price):
     return False
 
 
-def versions_are_same(version_obj: models.ApplicationVersion, raw_version: dict, datemode: int):
+def versions_are_same(
+    version_obj: models.ApplicationVersion, application_version: ApplicationVersionSchema
+) -> bool:
     """Checks if the given versions are to be considered equal"""
-
     return (
-        version_obj.application.tag == _get_tag_from_raw_version(raw_version)
-        and version_obj.valid_from
-        == datetime(*xlrd.xldate_as_tuple(float(raw_version["Valid from"]), datemode))
-        and prices_are_same(version_obj.price_standard, raw_version["Standard"])
-        and prices_are_same(version_obj.price_priority, raw_version["Priority"])
-        and prices_are_same(version_obj.price_express, raw_version["Express"])
-        and prices_are_same(version_obj.price_research, raw_version["Research"])
+        version_obj.application.tag == application_version.app_tag
+        and version_obj.valid_from == application_version.valid_from
+        and prices_are_same(version_obj.price_standard, application_version.standard)
+        and prices_are_same(version_obj.price_priority, application_version.priority)
+        and prices_are_same(version_obj.price_express, application_version.express)
+        and prices_are_same(version_obj.price_research, application_version.research)
     )
 
 
-def applications_are_same(application_obj: models.Application, raw_application: dict):
+def applications_are_same(application_obj: models.Application, application: ApplicationSchema):
     """Checks if the given applications are to be considered equal"""
 
-    return application_obj and application_obj.tag == _get_tag_from_raw_application(raw_application)
+    return application_obj and application_obj.tag == application.tag
```

Comment:
This one returns a boolean or None but it was not immediately obvious to me. I think it can be expanded slightly to be more readable. Also the function using this one already checks if application_obj exists

Reply:
Well spotted